### PR TITLE
Initial stuff required for migration to AWS LZ2

### PR DESCRIPTION
**Pull request by vyasworks**
What this PR does: 

* Lambda-ize main server app
* Update make file for terraform automation  
* Update build pipeline for server and cron lambdas 
* Impl CF deployment via terraform
* Terraform for Cloudfront and Lambda Deployment for main server
* Disable client app served via server. 

Tickets:

https://freshworks.atlassian.net/browse/ETS-474
https://freshworks.atlassian.net/browse/ETS-476
https://freshworks.atlassian.net/browse/ETS-480
https://freshworks.atlassian.net/browse/ETS-483

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -219,11 +197,6 @@ app.get(`${apiBaseUrl}/validate`,
 app.get(`${apiBaseUrl}/version`,
   (req, res) => res.json({ version: process.env.VERSION }));
 
-// Client app
```
Comment (vyasworks) [inline server/server.js:222]: We dont need this anymore, now that we have cloudfront hosting webapp

File: server/server.js [hunk 2]
```diff
@@ -18,27 +15,8 @@ const logger = require('./logger.js');
 const apiBaseUrl = '/api/v1';
 const app = express();
 
-app.use(helmet({
```
Comment (vyasworks) [inline server/server.js:21]: We dont need this anymore, now that we have cloudfront hosting webapp

File: makefile.old [hunk 3]
```diff
@@ -0,0 +1,120 @@
+#!make
```
Comment (vyasworks) [inline makefile.old:1]: This is the existing make file, moved to makefile.old for backwards compatability until migration is fully complete. 

The github actions are updated to use this file. For local, use this file.

File: terraform/dev/app.tf [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  is_ipv6_enabled     = true
+  default_root_object = "index.html"
+
+  default_cache_behavior {
+    allowed_methods        = ["GET", "HEAD", "OPTIONS"]
+    cached_methods         = ["GET", "HEAD", "OPTIONS"]
+    target_origin_id       = local.s3_origin_id
+    cache_policy_id        = data.aws_cloudfront_cache_policy.optimized.id
+    viewer_protocol_policy = "redirect-to-https"
+    # lambda_function_association {
```
Comment (vyasworks) [inline terraform/dev/app.tf:62]: Place holder for CSP

File: terraform/dev/app.tf [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+}
+
+resource "null_resource" "update_cf_app" {
+  provisioner "local-exec" {
+    command = <<EOF
```
Comment (vyasworks) [inline terraform/dev/app.tf:96]: This is where frontend deployment happens.

File: makefile [hunk 6]
```diff
@@ -1,29 +1,134 @@
 #!make
 
--include .env
+ENV_NAME ?= dev
 
-export $(shell sed 's/=.*//' .env)
+
+PROJECT_NAME = ets
```
Comment (arranfw) [inline makefile:6]:
```suggestion
export PROJECT_NAME = ets
```

This in addition to renaming the usages in docker-compose should fix `run-local`

File: terraform/dev/api.tf [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  memory_size      = 1024
+  timeout          = 10
+
+  environment {
+    variables = {
+      DB_SERVER   = "dummy"
+      DB_PORT     = "dummy"
+      DB_USER     = "dummy"
+      DB_PASSWORD = "dummy"
+      DB_NAME     = "dummy"
```
Comment (arranfw) [inline terraform/dev/api.tf:18]: I imagine this is WIP?

**Review by arranfw — COMMENTED** [review]
Couple comments, can't speak too much to the terraform unfortunately

File: makefile [hunk 8]
```diff
@@ -1,29 +1,134 @@
...
+LZ2_PROJECT = klwrig
+TERRAFORM_DIR = terraform/$(ENV_NAME)
+
+PROJECT_CODE = $(LZ2_PROJECT)-$(ENV_NAME)
+NAMESPACE = $(PROJECT_CODE)-$(ENV_NAME)
```
[the remark this answers is not in the document]
Comment (vyasworks) [reply makefile:11]: it would end up as `klwrig-dev`

[the remark this answers is not in the document]
Comment (arranfw) [reply makefile:11]: Ah okay, I interpreted it as evaluating to

```
NAMESPACE  = $(LZ2_PROJECT)-$(ENV_NAME)-$(ENV_NAME)
```

[the remark this answers is not in the document]
Comment (ashtonmeuser) [reply makefile:11]: I've got the same question. I don't really see how this avoids a repeated suffix.

File: makefile [hunk 9]
```diff
@@ -1,29 +1,134 @@
...
+
+apply: init
+	# Creating all AWS infrastructure.
+	@terraform -chdir=$(TERRAFORM_DIR) apply -auto-approve -input=false
+
+
+force-unlock: init
+	terraform -chdir=$(TERRAFORM_DIR) force-unlock $(LOCK_ID)
+
+destroy: init-tf
```
Comment (ashtonmeuser) [inline makefile:61]: Where is the `init-tf` target defined?

File: makefile [hunk 10]
```diff
@@ -1,29 +1,134 @@
...
+	@echo "Copy backend to temp folder to start build"
+	@echo "============================="
+	cp -r ./server ./terraform/build/server
+
+	@echo -e "\n\n\n============================="
+	@echo "Optimize node modules size"
+	@echo "============================="
+	npx modclean -n default:safe,default:caution -er -D terraform/build/server
+
+	curl -s https://gobinaries.com/tj/node-prune |  PREFIX=. sh
```
Comment (ashtonmeuser) [inline makefile:103]: This feels a little weird. Is this to trim down the zip file?

**Review by ashtonmeuser — APPROVED** [review]

File: makefile (hunk 10 above)
Comment (vyasworks) [reply makefile:103]: Yes. This is to redue unwanted junk, so lambda zips are trim.

File: makefile (hunk 9 above)
Comment (vyasworks) [reply makefile:61]: needs to be renamed to init.

File: makefile (hunk 6 above)
Comment (vyasworks) [reply makefile]: TODO mentioned in makefile

**Pull request merged**